Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -278,16 +294,15 @@ def network_create_engine_wrapper(self, pstate, initialdlstatus=None):
                 has_resume_data = resume_data and isinstance(resume_data, dict)
                 if has_resume_data:
                     atp["resume_data"] = lt.bencode(resume_data)
-                self._logger.info("%s %s", self.tdef.get_name_as_unicode(), dict((k, v)
-                                  for k, v in resume_data.iteritems() if k not in ['pieces', 'piece_priority', 'peers']) if has_resume_data else None)
             else:
                 atp["url"] = self.tdef.get_url() or "magnet:?xt=urn:btih:" + hexlify(self.tdef.get_infohash())
                 atp["name"] = self.tdef.get_name_as_unicode()
 
             self.handle = self.ltmgr.add_torrent(self, atp)
 
             if self.handle:
-                self.set_selected_files()
+                # if in share mode, don't change priority of the file
+                self.set_selected_files(share_mode=share_mode)
```

Comment:
Just wondering, we are passing `share_mode` around quite often. Maybe it's better to make it an attribute on `LibtorrentDownloadImpl` for easy access?